### ISSUE TITLE
GDAL 2.2.2 does not compile on Centos7,skylake, intel 2017b

### ISSUE THREAD

**Issue by hajgato** [easybuilders/easybuild-easyconfigs#5843]
`ceos.c` does not compile with `-O2` and `-xAVX-512` (on Skylake). it compiles with `-O1 -xAVX-512` and with `-O2 -xAVX2`. When I looked `mcpcom` with `strace`, it was doing nothing. (it was running for almost 23 hours that moment)

this patch solves the problem:
```
--- gdal-2.2.2/frmts/ceos2/GNUmakefile.org	2017-09-15 12:37:53.000000000 +0200
+++ gdal-2.2.2/frmts/ceos2/GNUmakefile	2018-02-19 21:42:14.864273222 +0100
@@ -12,6 +12,8 @@
 
 $(O_OBJ):	../raw/rawdataset.h
 
+../o/ceos.lo:	CFLAGS	+= -xAVX2
+
 clean:
 	rm -f *.o $(O_OBJ)
 
```

**Comment by boegel** [comment]
Just to confirm: I'm seeing the same issue when building `GDAL-2.2.2-intel-2017b-Python-2.7.14.eb` on Skylake/CentOS 7.4.1708; it seems like `icc` goes into lala-land when compiling `ceos.c`...

I'll try again with your patch, and also with `intel/2018a` to see if the issue persists...

**Comment by boegel** [comment]
@hajgato Building with your patch indeed fixes the problem. Adding `-xAVX2` isn't portable across architectures though, so maybe it's better to add `-O1` instead for `ceos.c`? That way we can apply the patch always, regardless of the architecture?

**Comment by boegel** [comment]
@hajgato I tried with `-O1` instead of `-xAVX2`, works fine too.

**Comment by hajgato** [comment]
@boegel Indeed the `-O1` solution is portable but in have the feeling that `-O2 -xAVX2` gives better performance. (we do see that AVX-512 not always give better performance). At the end `-O1` is the universal solution (this is why I only give the patches here). (unless we think about architecture-dependent patches....)

**Comment by boegel** [comment]
@hajgato We currently don't have a way to deal with architecture-dependent patches. I would expect that compiling a single file with `-O1 -xHost` rather than `-O2 -xHost` is rather limited...

Shall we open a PR to apply the patch to all recent GDAL easyconfigs using `intel` toolchain?

**Comment by boegel** [comment]
fixed in #5915, let's follow up there?
